packages/flutter_tools/lib/executable.dart

```diff
@@ -131,70 +131,78 @@ Future<Null> run(List<String> args, List<FlutterCommand> subCommands, {bool verb
     // Initialize the system locale.
     await intl.findSystemLocale();
 
-    return Chain.capture<Future<Null>>(() async {
+    Completer<Null> runCompleter = new Completer<Null>();
+    Chain.capture<Future<Null>>(() async {
       await runner.run(args);
       await _exit(0);
+      runCompleter.complete();
     }, onError: (dynamic error, Chain chain) {
-      if (error is UsageException) {
-        stderr.writeln(error.message);
-        stderr.writeln();
+      _handleToolError(error, chain, verbose, args).whenComplete(runCompleter.complete);
```

Comment:
No, because at this point we already reported the error to stderr, saved to the crash file, potentially sent a crash report, and set the exit code. We don't want to rethrow errors past this point, only signal that we're done running the sub-command, whether successfully or not. We can certainly rethink the approach, but I'd prefer changing it as a follow-up PR, if that's ok.